tests/unit/weak_ptr_test.cc

```diff
@@ -48,6 +48,15 @@ BOOST_AUTO_TEST_CASE(test_weak_ptr_is_reset) {
     BOOST_REQUIRE(!bool(wp));
 }
 
+BOOST_AUTO_TEST_CASE(test_const_weak_ptr) {
+    auto owning_ptr = std::make_unique<myclass>();
+
+    weak_ptr<const myclass> cwptr = const_cast<const myclass&>(*owning_ptr).weak_from_this();
```

Comment:
Please correct me if I'm wrong, but it seems the goal of this patch isn't to support the type weak_ptr<const T> or not only to support it - but rather specifically to support `weak_ptr<T const> weak_from_this() const` like the C++ standard supports (https://en.cppreference.com/w/cpp/memory/enable_shared_from_this/weak_from_this).

Reply:
Well, yes, the goal is to implement `weak_ptr<T const> weak_from_this() const` thus having const-weak-pointers. I'll try to rephrase the title